torch/distributions/distribution.py

```diff
@@ -18,7 +28,8 @@ def sample_n(self, n):
         Generates n samples or n batches of samples if the distribution parameters
         are batched.
         """
-        raise NotImplementedError
+        warnings.warn('sample_n will be deprecated. Use .sample((n,)) instead', PendingDeprecationWarning)
```

Comment:
Did you verify that this warning shows up? We've had some issues where if you mark them as `DeprecationWarning`s, then they are filtered out by default.

Reply:
That might depend on the client settings. In Pyro, we filter out DeprecationWarnings (but emit once), since we found some problematic dependencies which were using some deprecated methods of their own dependencies. 

@alicanb - You could test that the warning is thrown in by using the [catch warnings context manager](https://docs.python.org/3/library/warnings.html#testing-warnings)